### PR TITLE
Rename CalculateHeadersWork to CalculateClaimedHeadersWork

### PR DESCRIPTION
**Pull request by instagibbs**
And clean up some comments. Confusion about what this is doing seems to be a running theme:

https://github.com/bitcoin/bitcoin/pull/29549#discussion_r1511113344

https://github.com/bitcoin/bitcoin/pull/27278#discussion_r1141510303

### CODE REVIEW

**Review by pablomartin4btc — COMMENTED** [review]
ACK eb7cc9fd2140df77acc6eb42004cf45b260bc629

File: src/net_processing.cpp [hunk 1]
```diff
@@ -4744,8 +4744,8 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             // cs_main in ProcessNewBlock is fine.
             mapBlockSource.emplace(hash, std::make_pair(pfrom.GetId(), true));
 
-            // Check work on this block against our anti-dos thresholds.
-            if (prev_block && prev_block->nChainWork + CalculateHeadersWork({pblock->GetBlockHeader()}) >= GetAntiDoSWorkThreshold()) {
+            // Check claimed work on this block against our anti-dos thresholds.
+            if (prev_block && prev_block->nChainWork + CalculateClaimedHeadersWork({pblock->GetBlockHeader()}) >= GetAntiDoSWorkThreshold()) {
                 min_pow_checked = true;
```
Comment (0xB10C) [inline src/net_processing.cpp:4749]: I also found `min_pow_checked` confusing. At least I read "PoW checked" at first.

**Review by 0xB10C — APPROVED** [review]
ACK eb7cc9fd2140df77acc6eb42004cf45b260bc629

I misinterpreted `CalculateHeadersWork` - `CalculateClaimedHeadersWork` is a bit clearer.

**Review by dergoegge — APPROVED** [review]
ACK eb7cc9fd2140df77acc6eb42004cf45b260bc629

File: src/net_processing.cpp (hunk 1 above)
Comment (instagibbs) [reply src/net_processing.cpp:4749]: agreed it's also confusing, but it's also called that down the call stack for a bit, so I kept scope to here

**Review by BrandonOdiwuor — APPROVED** [review]
ACK eb7cc9fd2140df77acc6eb42004cf45b260bc629

**Comment by achow101** [comment]
ACK eb7cc9fd2140df77acc6eb42004cf45b260bc629

**Pull request merged**